prime-router/src/test/kotlin/fhirengine/utils/FHIRBundleHelpersTests.kt

```diff
@@ -777,14 +781,65 @@ class FHIRBundleHelpersTests {
         code.addCoding(Coding("system", "some-unmapped-code", "display"))
         entry.setCode(code)
 
-        val logs = entry.addMappedCondition(metadata)
+        val logs = entry.addMappedConditions(metadata)
         assertThat(logs.size).isEqualTo(0)
 
         val extension = code.coding.first().extension.first()
         assertThat(extension.url).isEqualTo(conditionCodeExtensionURL)
         assertThat((extension.value as? Coding)?.code).isEqualTo("6142004")
     }
 
+    @Test
+    fun `Ensure a mapped observation is stamped with all condition codes if there are multiple`() {
+        val metadata = Metadata(UnitTestUtils.simpleSchema)
+
+        metadata.lookupTableStore += mapOf(
+            "observation-mapping" to LookupTable(
+                "observation-mapping",
+                listOf(
+                    listOf(
+                        ObservationMappingConstants.TEST_CODE_KEY,
+                        ObservationMappingConstants.CONDITION_CODE_KEY,
+                        ObservationMappingConstants.CONDITION_CODE_SYSTEM_KEY,
+                        ObservationMappingConstants.CONDITION_NAME_KEY
+                    ),
+                    listOf(
+                        "80382-5",
+                        "6142004",
+                        "SNOMEDCT",
+                        "Influenza (disorder)"
+                    ),
+                    listOf(
+                        "260373001",
+                        "Some Condition Code",
+                        "Condition Code System",
+                        "Condition Name"
+                    )
+                )
+            )
+        )
+
+        val entry = Observation()
+        val code = CodeableConcept()
+        code.addCoding(Coding("system", "80382-5", "display"))
```

Comment:
This might be me not grokking the test/fix here, but I thought we were handling situations where the same test code maps to multiple conditions? i.e. `"80382-5"` maps to Influenza (disorder) and Influenza A

Reply:
Derp; nice catch!